pkg/kube/object_patch/patch.go

```diff
@@ -0,0 +1,344 @@
+		case JQPatch:
+			operationError = o.JQPatchObject(spec.JQFilter, spec.ApiVersion, spec.Kind, spec.Namespace, spec.Name, spec.Subresource)
+		case MergePatch:
+			jsonMergePatch, err := json.Marshal(spec.MergePatch)
+			if err != nil {
+				applyErrors = multierror.Append(applyErrors, err)
+			}
+
+			operationError = o.MergePatchObject(jsonMergePatch, spec.ApiVersion, spec.Kind, spec.Namespace, spec.Name, spec.Subresource)
+		case JSONPatch:
+			jsonJsonPatch, err := json.Marshal(spec.MergePatch)
+			if err != nil {
+				applyErrors = multierror.Append(applyErrors, err)
+			}
+
+			operationError = o.JSONPatchObject(jsonJsonPatch, spec.ApiVersion, spec.Kind, spec.Namespace, spec.Name, spec.Subresource)
+		}
+
+		if operationError != nil {
+			applyErrors = multierror.Append(applyErrors, operationError)
+		}
+	}
+
+	//goland:noinspection GoNilness
+	return applyErrors.ErrorOrNil()
+}
+
+func unmarshalFromJsonOrYaml(specs []byte) ([]OperationSpec, error) {
+	fromJsonSpecs, err := unmarshalFromJson(specs)
+	if err != nil {
+		return unmarshalFromYaml(specs)
+	}
+
+	return fromJsonSpecs, nil
+}
+
+func unmarshalFromJson(jsonSpecs []byte) ([]OperationSpec, error) {
+	var specSlice []OperationSpec
+
+	dec := json.NewDecoder(bytes.NewReader(jsonSpecs))
+	for {
+		var doc OperationSpec
+		err := dec.Decode(&doc)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		specSlice = append(specSlice, doc)
+	}
+
+	return specSlice, nil
+}
+
+func unmarshalFromYaml(yamlSpecs []byte) ([]OperationSpec, error) {
+	var specSlice []OperationSpec
+
+	dec := yaml.NewDecoder(bytes.NewReader(yamlSpecs))
+	for {
+		var doc OperationSpec
+		err := dec.Decode(&doc)
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		specSlice = append(specSlice, doc)
+	}
+
+	return specSlice, nil
+}
+
+func (o *ObjectPatcher) CreateObject(object *unstructured.Unstructured, namespace, subresource string) error {
+	if object == nil {
+		return fmt.Errorf("cannot create empty object")
+	}
+
+	apiVersion := object.GetAPIVersion()
+	kind := object.GetKind()
+
+	gvk, err := o.kubeClient.GroupVersionResource(apiVersion, kind)
+	if err != nil {
+		return err
+	}
+
+	_, err = o.kubeClient.Dynamic().Resource(gvk).Namespace(namespace).Create(object, metav1.CreateOptions{}, generateSubresources(subresource)...)
+
+	return err
+}
+
+func (o *ObjectPatcher) CreateOrUpdateObject(object *unstructured.Unstructured, namespace, subresource string) error {
+	if object == nil {
+		return fmt.Errorf("cannot create empty object")
+	}
+
+	apiVersion := object.GetAPIVersion()
+	kind := object.GetKind()
+
+	gvk, err := o.kubeClient.GroupVersionResource(apiVersion, kind)
+	if err != nil {
+		return err
+	}
+
+	_, err = o.kubeClient.Dynamic().Resource(gvk).Namespace(namespace).Create(object, metav1.CreateOptions{}, generateSubresources(subresource)...)
+	if err != nil {
+		if errors.IsAlreadyExists(err) {
```

Comment:
I think you don't have to check that `err != nil` before this check.

Reply:
Indeed!